src/providers/Auth/Auth.provider.jsx

```diff
@@ -1,42 +1,45 @@
-import React, { useState, useEffect, useContext, useCallback } from 'react';
+import React, { useEffect, useContext, useState, useCallback } from 'react';
+import { USERS } from '../../constants';
 
-import { AUTH_STORAGE_KEY } from '../../utils/constants';
-import { storage } from '../../utils/storage';
-
-const AuthContext = React.createContext(null);
+const UserContext = React.createContext(null);
 
 function useAuth() {
-  const context = useContext(AuthContext);
+  const context = useContext(UserContext);
   if (!context) {
-    throw new Error(`Can't use "useAuth" without an AuthProvider!`);
+    throw new Error(`Can't use "useAuth" without an AuthContext!`);
```

Comment:
Change it to `AuthProvider` is the only way that the context would be null. If you use the hook without having the Provider as a parent in the tree from the component.